builders/build/canonical-contracts/precompiles/batch.md

```diff
@@ -0,0 +1,225 @@
+
+The **SimpleMessage** contract will appear in the list of **Deployed Contracts**.
+
+### Send Native Currency via Precompile {: #send-native-currency-via-precompile }
+
+Sending native currency with the batch precompile is more involved than pressing a few buttons in Remix or Metamask. 
+
+Transactions have a value field to specify the amount of native currency being sent with it. In Remix, this is represented by the **VALUE** input in the **DEPLOY & RUN TRANSACTIONS** tab. However, for the batch precompile, this data is provided within the *value* array input of the batch functions.
+
+Try transferring native currency to two wallets of your choice via the batch precompile on Moonbase Alpha:
+
+1. Make sure that you have at least 0.5 DEV in your connected wallet
+2. Expand the Batch contract under **Deployed Contracts**
+3. Expand the **batchAll** function
+4. For the *to* input, insert your addresses in the following format: `["0xADDRESS1", "0xADDRESS2"]`, where 0xADDRESS1 is the address of the first wallet of your choice and 0xADDRESS2 is the address of the second wallet of your choice
+5. For the value input, insert the amount you wish to transfer in wei for each address. For example, `["100000000000000000", "200000000000000000"]` will transfer 0.1 DEV to 0xADDRESS1 and 0.2 DEV to 0xADDRESS2
+6. For both of the remaining *call_data* and *gas_limit* inputs, insert `[]`. Call data and gas limit are not a concern for transferring native currency
+7. Press **transact**
+8. Press **Confirm** in the Metamask extension to confirm the transaction
+
+![Send Batch Transfer](/images/builders/build/canonical-contracts/precompiles/batch/batch-4.png)
+
+Once the transaction is complete, be sure to check both of the accounts' balances, either in Metamask or in a [block explorer](/builders/get-started/explorers/){target=_blank}. Congratulations! You've now sent a batched transfer via the Batch precompile.
+
+!!! note
+     Typically if you wanted to send the native currency to or through a contract, you would have to set the value within the overall transaction and interact with a payable function. However, since the Batch precompile interacts directly with Substrate code, this is not necessary.
+
+
+### Finding a Contract Interaction's Call Data {: #finding-a-contract-interactions-call-data } 
+
+Visual interfaces like Remix and handy libraries like ethers.js hide the way that ethereum transactions interact with solidity smart contracts. The name and input types of a function are hashed into a [function selector](https://docs.soliditylang.org/en/latest/abi-spec.html#function-selector-and-argument-encoding){target=_blank} and the input data is encoded. These two pieces are then combined and sent as the transaction's call data. To send a subtransaction within a batch transaction, the sender to know its call data beforehand. 
+
+Try finding a transaction's call data using Remix:
+
+1. Expand the SimpleMessage contract under **Deployed Contracts**
```

Comment:
I followed the same format as the ERC20 precompile, which didn't apply anything to the names of the contracts

Reply:
I'll replace it with SimpleContract so that I don't have to take the screenshots again!

I reference the contract a lot in the text. Should I make them `SimpleContract.sol` as well?

Reply:
sounds good and yes please!